scripts/gui/microbe_hud.mjs

```diff
@@ -481,27 +640,127 @@ function hideWinText(){
+
+        for (const barValue of barValues) {
+            barValue.style.left = dataToChange.valueLeft + "px";
+        }
+    }
+}
+
+//! Expand panel function
+function onExpandPanelClicked(panelToChange, dataToChange) {
+
+
+    $("#Panels").animate({height: $("#Panels").height() + dataToChange.height + "px"}, 300);
+
+    // Change buttons status
+    panelToChange.querySelector(".compressPanel").style.backgroundImage =
+        "url('../../Textures/gui/bevel/compressPanel.png')";
+    panelToChange.querySelector(".expandPanel").style.backgroundImage =
+        "url('../../Textures/gui/bevel/expandPanelActive.png')";
+
+    panelToChange.style.backgroundImage = dataToChange.background;
+    panelToChange.style.height = panelToChange.offsetHeight + dataToChange.height + "px";
+
+    const rows = panelToChange.querySelectorAll(".row");
+
+    for(const row of rows) {
+        const bars = row.getElementsByClassName(dataToChange.bar);
+        const title = row.getElementsByClassName(dataToChange.title);
+        const barValues = row.getElementsByClassName(dataToChange.value);
+
+        for (const bar of bars) {
+            bar.style.display = "block";
+            bar.style.marginBottom = "4px";
+            bar.style.marginTop = "6px";
+            bar.style.width = bar.offsetWidth + dataToChange.width + "px";
+            bar.style.marginLeft = dataToChange.leftMargin + "px";
+        }
+
+        for (const tit of title) {
+            tit.style.visibility = "visible";
+        }
+
+        for (const barValue of barValues) {
+            barValue.style.left = dataToChange.valueLeft + "px";
+        }
+    }
+}
+
 //! Updates the GUI bars
 //! values needs to be an object with properties set with values for everything
 function updateMicrobeHUDBars(values){
+    // The bars
     document.getElementById("microbeHUDPlayerHitpoints").textContent =
         values.hitpoints;
     document.getElementById("microbeHUDPlayerMaxHitpoints").textContent =
         values.maxHitpoints;
-    document.getElementById("microbeHUDPlayerHitpointsBar").style.width =
-        common.barHelper(values.hitpoints, values.maxHitpoints);
-
-    // TODO: remove this debug code
-    document.getElementById("microbeHUDPlayerATP").textContent =
-        values.compoundATP.toFixed(1);
 
-    // The bars
     document.getElementById("microbeHUDPlayerATPCompound").textContent =
          values.compoundATP.toFixed(1);
     document.getElementById("microbeHUDPlayerATPMax").textContent =
          values.ATPMax;
-    document.getElementById("microbeHUDPlayerATPBar").style.width =
-         common.barHelper(values.compoundATP, values.ATPMax);
+
+    const valueAtp = common.barHelper(values.compoundATP, values.ATPMax).replace("%", "");
+    const valueHp = common.barHelper(values.hitpoints, values.maxHitpoints).replace("%", "");
+
+    const circles = document.querySelectorAll("#circleBars");
+
+    // ! instead of using totalProgress var, two hardCoded value are used
```

Comment:
Doxygen style comments only work on functions, classes etc. Inline comments should use the normal comment style.